### PR TITLE
extra ";" generation has been taken care for Bang command

### LINKED ISSUE #851: [BUG]: `!set ...` commands break Scala functional tests
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Category of Bug / Issue

Other

### Current Behavior

```
 === Unexpected output (expected vs actual) ===
!-- snowsql command:!'set exit_on_error = true';   --!set exit_on_error = true;;
```
image

### Expected Behavior

There must be no errors

### Steps To Reproduce

run `SnowflakeAcceptanceSuite`

### Relevant log output or Exception details

_No response_

### Sample Query

_No response_

### Operating System

macOS

### Version

latest via Databricks CLI

### PR DESCRIPTION
**Pull request by sriram251-code**
fixed #851

### CODE REVIEW

File: core/src/main/scala/com/databricks/labs/remorph/generators/sql/LogicalPlanGenerator.scala [hunk 1]
```diff
@@ -8,7 +8,18 @@ class LogicalPlanGenerator(val expr: ExpressionGenerator, val explicitDistinct:
     extends Generator[ir.LogicalPlan, String] {
 
   override def generate(ctx: GeneratorContext, tree: ir.LogicalPlan): String = tree match {
-    case b: ir.Batch => b.children.map(generate(ctx, _)).mkString("", ";\n", ";")
+    case b: ir.Batch =>
+      b.children
```
Comment (nfx) [inline core/src/main/scala/com/databricks/labs/remorph/generators/sql/LogicalPlanGenerator.scala:12]: refactor this branch into `batch(ctx, b)` private method

File: core/src/main/scala/com/databricks/labs/remorph/generators/sql/LogicalPlanGenerator.scala [hunk 2]
```diff
@@ -8,7 +8,18 @@ class LogicalPlanGenerator(val expr: ExpressionGenerator, val explicitDistinct:
...
+        .map {
+          case ir.UnresolvedCommand(text) =>
+            if (text.endsWith(";")) {
+              text
+            } else {
+              text + ";";
+            }
+          case query => s"${generate(ctx, query);}"
```
Comment (nfx) [inline core/src/main/scala/com/databricks/labs/remorph/generators/sql/LogicalPlanGenerator.scala:20]:
```suggestion
          case query => s"${generate(ctx, query)};"
```

File: core/src/test/scala/com/databricks/labs/remorph/transpilers/SnowflakeToDatabricksTranspilerTest.scala [hunk 3]
```diff
@@ -9,8 +9,8 @@ class SnowflakeToDatabricksTranspilerTest extends AnyWordSpec with TranspilerTes
   "snowsql commands" should {
 
     "transpile BANG" in {
-      "!set error_flag = true" transpilesTo "--!set error_flag = true;"
-      "!define tablename=CENUSTRACKONE" transpilesTo "--!define tablename=CENUSTRACKONE;"
+      "!set error_flag = true;" transpilesTo "--!set error_flag = true;"
+      "!define tablename=CENUSTRACKONE;" transpilesTo "--!define tablename=CENUSTRACKONE;"
```
Comment (nfx) [inline core/src/test/scala/com/databricks/labs/remorph/transpilers/SnowflakeToDatabricksTranspilerTest.scala:13]: make format the same on both python and scala side. this PR has to have +3/-3 functional test results

File: core/src/main/scala/com/databricks/labs/remorph/generators/sql/LogicalPlanGenerator.scala (hunk 1 above)
Comment (sriram251-code) [reply core/src/main/scala/com/databricks/labs/remorph/generators/sql/LogicalPlanGenerator.scala:14]: done

File: core/src/main/scala/com/databricks/labs/remorph/generators/sql/LogicalPlanGenerator.scala (hunk 2 above)
Comment (sriram251-code) [reply core/src/main/scala/com/databricks/labs/remorph/generators/sql/LogicalPlanGenerator.scala]: done

File: core/src/test/scala/com/databricks/labs/remorph/transpilers/SnowflakeToDatabricksTranspilerTest.scala (hunk 3 above)
Comment (sriram251-code) [reply core/src/test/scala/com/databricks/labs/remorph/transpilers/SnowflakeToDatabricksTranspilerTest.scala]: I have added +3/-3 test cases for antlr. should i add the same test cases for sqlglot?

**Comment by sriram251-code** [comment]
@nfx As this branch has been created from fork branch hence the Coverage tests results are not getting published. So created another PR  https://github.com/databrickslabs/remorph/pull/858 on which the coverage test results are getting published.All the review comments has been addressed in this new PR.

**Pull request closed without merging**